prow/plugins/config.go

```diff
@@ -1786,9 +1786,58 @@ func (c *Configuration) mergeFrom(other *Configuration) error {
 		errs = append(errs, fmt.Errorf("failed to merge .bugzilla from supplemental config: %w", err))
 	}
 
+	if err := c.mergeApproveFrom(other.Approve, org, repo); err != nil {
+		errs = append(errs, fmt.Errorf("failed to merge .approve from supplemental config: %w", err))
+	}
+
 	return utilerrors.NewAggregate(errs)
 }
 
+func (c *Configuration) mergeApproveFrom(other []Approve, org, repo string) error {
+	if len(other) == 0 {
+		return nil
+	}
+
+	// If the supplemental configuration points to a specific org/repo, we need to validate that there is
+	// only one approve configuration that holds that org/repo value in the `repos` field.
+	if org != "" && repo != "" {
+		if len(other) > 1 {
+			return fmt.Errorf("supplemental config for %s/%s must have only 1 approve plugin configuration", org, repo)
+		}
+		for _, approve := range other {
+			if len(approve.Repos) != 1 || (len(approve.Repos) == 1 && approve.Repos[0] != fmt.Sprintf("%s/%s", org, repo)) {
+				return fmt.Errorf("the repos field must hold the %s/%s value only", org, repo)
+			}
+		}
+	}
+
+	// If the supplemental configuration points to an organization, we need to validate that each of the configurations holds
+	// values in the repo field that belong to that organization.
+	if org != "" && repo == "" {
+		var errs []error
+		for i, approve := range other {
+			for j, orgRepo := range approve.Repos {
+				orgValue := strings.Split(orgRepo, "/")[0]
+				if orgValue != org {
+					errs = append(errs, fmt.Errorf("approve[%d].repos[%d]: value %s doesn't belong in %s organization", i, j, orgRepo, org))
+				}
+			}
+		}
+		if len(errs) > 0 {
+			return utilerrors.NewAggregate(errs)
+		}
+	}
+
+	if len(c.Approve) == 0 {
+		c.Approve = other
+		return nil
+	}
+
+	c.Approve = append(c.Approve, other...)
```

Comment:
I don't think this is the correct implementation. Since the `[]Approve` slice is order-dependent, we must make sure that any org-level configuration comes *later* in this list than any repo-level configurations inside of that org, that way the repo-level configurations override the org-level ones.

Reply:
well, we validate that for repo-level config this slice will have only 1 value. It can have multiple values only for org-level configs.

Reply:
In this case, we will need to re-order the configuration after we merge all of them.

Reply:
Well, it seems that we already do that. We search first for a configuration for the specific repo and if we won't find it we search again for the org only. See: https://github.com/kubernetes/test-infra/blob/master/prow/plugins/config.go#L751-L770